[CORE-7129] `rptest`: deflake `wait_for_internal_scrub()`

WillemKauf opened the pull request:
## Backports Required

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

* none

WillemKauf committed:
`rptest`: bump timeout in `wait_for_internal_scrub`

This was timing out occasionally in CI.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/63067-->
<!--49b5e233-2455-4383-8525-b2dc5ec8469f-->
<!--title-->
## Retry command for [Build#63067](https://buildkite.com/redpanda/redpanda/builds/63067)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/retention_policy_test.py::ShadowIndexingLocalRetentionTest.test_local_time_based_retention_is_overridden@{"cloud_storage_type":2,"local_retention_ms":-1}
tests/rptest/tests/retention_policy_test.py::ShadowIndexingCloudRetentionTest.test_cloud_size_based_retention_application@{"cloud_storage_type":1}
tests/rptest/tests/retention_policy_test.py::ShadowIndexingCloudRetentionTest.test_cloud_retention_deleted_segments_count@{"cloud_storage_type":1}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--06f67a6e-824c-465d-bc9d-83edef977b85-->
<!--TEST RESULTS-->
<!--41875cc1-f2e2-4939-bb57-fa4780e9b4d8-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/63067">build#63067</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| cloud_storage_rpfixture.cloud_storage_rpfixture | unit | https://buildkite.com/redpanda/redpanda/builds/63067#019590a5-94d2-4f57-8382-09bfc5a6564a | FAIL | 0/2 |
| cloud_storage_rpfixture.cloud_storage_rpfixture | unit | https://buildkite.com/redpanda/redpanda/builds/63067#019590a5-94d3-49e8-9ab9-ec98718000f9 | FAIL | 0/2 |
| cloud_storage_rpfixture.cloud_storage_rpfixture | unit | https://buildkite.com/redpanda/redpanda/builds/63067#0195914b-4cf3-4401-8143-63f4f240d6df | FAIL | 0/2 |
| rptest.tests.archive_retention_test.CloudArchiveRetentionTest.test_delete.cloud_storage_type=CloudStorageType.ABS.retention_type=retention.bytes | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e68-4818-9af2-4cbf3a9573db | FLAKY | 1/5 |
| rptest.tests.archive_retention_test.CloudArchiveRetentionTest.test_delete.cloud_storage_type=CloudStorageType.ABS.retention_type=retention.bytes | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af67-4456-b009-d80c39f2260d | FLAKY | 1/2 |
| rptest.tests.archive_retention_test.CloudArchiveRetentionTest.test_delete.cloud_storage_type=CloudStorageType.ABS.retention_type=retention.bytes | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#0195918f-a90d-4ea4-9811-6b7111166262 | FLAKY | 1/3 |
| rptest.tests.archive_retention_test.CloudArchiveRetentionTest.test_delete.cloud_storage_type=CloudStorageType.ABS.retention_type=retention.ms | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e69-4eb3-a739-3521bf3c1449 | FAIL | 0/20 |
| rptest.tests.archive_retention_test.CloudArchiveRetentionTest.test_delete.cloud_storage_type=CloudStorageType.ABS.retention_type=retention.ms | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af67-4f05-ad1e-369142933546 | FLAKY | 1/12 |
| rptest.tests.archive_retention_test.CloudArchiveRetentionTest.test_delete.cloud_storage_type=CloudStorageType.ABS.retention_type=retention.ms | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#0195918f-a90e-417b-b565-7fbc47816ee5 | FAIL | 0/20 |
| rptest.tests.archive_retention_test.CloudArchiveRetentionTest.test_delete.cloud_storage_type=CloudStorageType.S3.retention_type=retention.bytes | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e67-4806-a4ca-f893453ab852 | FLAKY | 1/11 |
| rptest.tests.archive_retention_test.CloudArchiveRetentionTest.test_delete.cloud_storage_type=CloudStorageType.S3.retention_type=retention.bytes | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af66-4c07-8458-48d6926387dc | FLAKY | 1/8 |
| rptest.tests.archive_retention_test.CloudArchiveRetentionTest.test_delete.cloud_storage_type=CloudStorageType.S3.retention_type=retention.ms | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af66-4db7-8d59-3ba78a98a466 | FAIL | 0/20 |
| rptest.tests.cloud_retention_test.CloudRetentionTest.test_cloud_retention.max_consume_rate_mb=20.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e69-4eb3-a739-3521bf3c1449 | FLAKY | 1/2 |
| rptest.tests.cloud_retention_test.CloudRetentionTest.test_gc_entire_manifest.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e68-4818-9af2-4cbf3a9573db | FAIL | 0/20 |
| rptest.tests.cloud_retention_test.CloudRetentionTest.test_gc_entire_manifest.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#0195918f-a90d-4ea4-9811-6b7111166262 | FAIL | 0/20 |
| rptest.tests.cloud_retention_test.CloudRetentionTest.test_gc_entire_manifest.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e69-4eb3-a739-3521bf3c1449 | FAIL | 0/20 |
| rptest.tests.cloud_retention_test.CloudRetentionTest.test_gc_entire_manifest.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#0195918f-a90e-417b-b565-7fbc47816ee5 | FAIL | 0/20 |
| rptest.tests.cloud_retention_test.CloudRetentionTimelyGCTest.test_retention_with_node_failures.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e67-4806-a4ca-f893453ab852 | FAIL | 0/20 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=True.params=.cancellation.dir.in.stage.preparing.use_alias.True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af66-4c07-8458-48d6926387dc | FLAKY | 1/2 |
| rptest.tests.datalake.datalake_e2e_test.DatalakeE2ETests.test_topic_lifecycle.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af67-4456-b009-d80c39f2260d | FLAKY | 1/2 |
| rptest.tests.datalake.rest_catalog_connection_test.RestCatalogConnectionTest.test_redpanda_connection_to_rest_catalog.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af67-4f05-ad1e-369142933546 | FLAKY | 1/2 |
| rptest.tests.delete_records_test.DeleteRecordsTest.test_delete_records_bounds_checking.cloud_storage_enabled=True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af66-4c07-8458-48d6926387dc | FLAKY | 1/2 |
| rptest.tests.delete_records_test.DeleteRecordsTest.test_delete_records_with_transactions.cloud_storage_enabled=True | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af66-4c07-8458-48d6926387dc | FLAKY | 1/2 |
| rptest.tests.e2e_shadow_indexing_test.ShadowIndexingWhileBusyTest.test_create_or_delete_topics_while_busy.short_retention=True.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e68-4818-9af2-4cbf3a9573db | FLAKY | 1/2 |
| rptest.tests.internal_topic_protection_test.InternalTopicProtectionLargeClusterTest.test_consumer_offset_topic | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af66-4db7-8d59-3ba78a98a466 | FLAKY | 1/2 |
| rptest.tests.partition_movement_test.SIPartitionMovementTest.test_shadow_indexing.num_to_upgrade=0.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e67-4806-a4ca-f893453ab852 | FLAKY | 1/2 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_retention_deleted_segments_count.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e69-4eb3-a739-3521bf3c1449 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_retention_deleted_segments_count.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af66-4db7-8d59-3ba78a98a466 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_retention_deleted_segments_count.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#0195918f-a90e-417b-b565-7fbc47816ee5 | FLAKY | 1/9 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_retention_deleted_segments_count.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e67-4806-a4ca-f893453ab852 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_retention_deleted_segments_count.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af67-4456-b009-d80c39f2260d | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_size_based_retention.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af67-4f05-ad1e-369142933546 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_size_based_retention.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e68-4818-9af2-4cbf3a9573db | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_size_based_retention.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af66-4c07-8458-48d6926387dc | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_size_based_retention.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#0195918f-a90d-4ea4-9811-6b7111166262 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_size_based_retention_application.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e69-4eb3-a739-3521bf3c1449 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_size_based_retention_application.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af66-4db7-8d59-3ba78a98a466 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_size_based_retention_application.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#0195918f-a90e-417b-b565-7fbc47816ee5 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_size_based_retention_application.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e67-4806-a4ca-f893453ab852 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingCloudRetentionTest.test_cloud_size_based_retention_application.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af67-4456-b009-d80c39f2260d | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingLocalRetentionTest.test_local_time_based_retention_is_overridden.local_retention_ms=-1.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e67-4806-a4ca-f893453ab852 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingLocalRetentionTest.test_local_time_based_retention_is_overridden.local_retention_ms=-1.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af67-4456-b009-d80c39f2260d | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingLocalRetentionTest.test_local_time_based_retention_is_overridden.local_retention_ms=-1.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af67-4f05-ad1e-369142933546 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingLocalRetentionTest.test_local_time_based_retention_is_overridden.local_retention_ms=3600000.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e68-4818-9af2-4cbf3a9573db | FLAKY | 1/16 |
| rptest.tests.retention_policy_test.ShadowIndexingLocalRetentionTest.test_local_time_based_retention_is_overridden.local_retention_ms=3600000.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af66-4c07-8458-48d6926387dc | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingLocalRetentionTest.test_local_time_based_retention_is_overridden.local_retention_ms=3600000.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#0195918f-a90d-4ea4-9811-6b7111166262 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingLocalRetentionTest.test_local_time_based_retention_is_overridden.local_retention_ms=3600000.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e69-4eb3-a739-3521bf3c1449 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingLocalRetentionTest.test_local_time_based_retention_is_overridden.local_retention_ms=3600000.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af66-4db7-8d59-3ba78a98a466 | FAIL | 0/20 |
| rptest.tests.retention_policy_test.ShadowIndexingLocalRetentionTest.test_local_time_based_retention_is_overridden.local_retention_ms=3600000.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#0195918f-a90e-417b-b565-7fbc47816ee5 | FAIL | 0/20 |
| rptest.tests.timequery_test.TimeQueryTest.test_timequery_with_trim_prefix.cloud_storage=True.spillover=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590e8-4e68-4818-9af2-4cbf3a9573db | FLAKY | 1/4 |
| rptest.tests.timequery_test.TimeQueryTest.test_timequery_with_trim_prefix.cloud_storage=True.spillover=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#019590fc-af67-4f05-ad1e-369142933546 | FLAKY | 1/4 |
| rptest.tests.timequery_test.TimeQueryTest.test_timequery_with_trim_prefix.cloud_storage=True.spillover=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63067#0195918f-a90d-4ea4-9811-6b7111166262 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/63107">build#63107</a></summary>


| test_id                                                                                                                                                                | test_kind   | job_url                                                                                   | test_status   | passed   |
|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.e2e_shadow_indexing_test.ShadowIndexingWhileBusyTest.test_create_or_delete_topics_while_busy.short_retention=True.cloud_storage_type=CloudStorageType.ABS | ducktape    | https://buildkite.com/redpanda/redpanda/builds/63107#01959317-c72c-420e-9e7c-1956ebe41d70 | FLAKY         | 10/11    |
| rptest.tests.e2e_shadow_indexing_test.ShadowIndexingWhileBusyTest.test_create_or_delete_topics_while_busy.short_retention=True.cloud_storage_type=CloudStorageType.ABS | ducktape    | https://buildkite.com/redpanda/redpanda/builds/63107#01959317-c72c-4bfd-a0b4-302d1156be85 | FLAKY         | 10/11    |
| rptest.tests.e2e_shadow_indexing_test.ShadowIndexingWhileBusyTest.test_create_or_delete_topics_while_busy.short_retention=True.cloud_storage_type=CloudStorageType.ABS | ducktape    | https://buildkite.com/redpanda/redpanda/builds/63107#01959317-c72c-4ffa-9485-43a8652e5a74 | FLAKY         | 10/11    |
</details>
<!--/body-->

WillemKauf committed:
`rptest`: reset scrubbing metadata before querying anomalies

Resetting the scrubbing metadata for every partition and then
querying them later can be race-y with retention enforcement.

To deflake the test, reset scrubbing metadata right before querying
its cloud storage anomalies.

WillemKauf commented:
/ci-repeat 10
skip-units
dt-repeat=10
tests/rptest/tests/e2e_shadow_indexing_test.py::ShadowIndexingWhileBusyTest.test_create_or_delete_topics_while_busy

WillemKauf commented:
This requires fixes in core.
